File: OOMTest-Xamarin/OOMTest/ImageViewPagerAdapter.cs
using System;
using Android.Support.V4.App;
namespace OOMTest
{
	public class ImageViewPagerAdapter : FragmentStatePagerAdapter
	{
		public ImageViewPagerAdapter(FragmentManager fm) 
			: base(fm)
		{
		}

		public override int Count {
			get {
				return 20;
			}
		}

		public override Fragment GetItem(int position)
		{
			return new ImageFragment();
		}
	}
}

File: OOMTest-Android/app/src/main/java/com/tobishiba/oomtest/ImageViewPagerAdapter.java
package com.tobishiba.oomtest;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

/**
 * Created by tobiasbuchholz on 12/09/16
 */
public class ImageViewPagerAdapter extends FragmentStatePagerAdapter {
    public ImageViewPagerAdapter(final FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(final int position) {
        return new ImageFragment();
    }

    @Override
    public int getCount() {
        return 20;
    }
}
